include/mega/types.h

```diff
@@ -311,7 +311,42 @@ typedef list<HttpReqCommandPutFA*> putfa_list;
 // map a FileFingerprint to the transfer for that FileFingerprint
 typedef map<FileFingerprint*, Transfer*, FileFingerprintCmp> transfer_map;
 
-typedef deque<Transfer*> transfer_list;
+template <class T, class E>
+class deque_with_lazy_bulk_erase
```

Comment:
Yes I tried to keep them small for that reason (and in general).  But the main constraint is that it's a template class, so those functions have to be in the header.   I agree that types.h is growing and perhaps has too much in it now - shall we split out a separate header, perhaps helperClasses.h and put these inline classes in it?  thanks

Reply:
Oh, you are correct! I didn't realize about the template class restriction, sorry. Nevertheless, it's a good idea to split out this file like that. Let me add a comment to the ticket to split out the `megaapi_impl.cpp`, so we do both at once rather than in this PR ;)